### ISSUE TITLE
Current validation fails for Python 3, succeeds for Python 2

### ISSUE THREAD

**Issue by Ichoran** [openworm/tracker-commons#44]
@MichaelCurrie - Not sure what's going on here.  I merged my Julia implementation anyway, but I think the problem happened earlier.  Maybe when I added the `intermediate.wcon` file?

**Comment by MichaelCurrie** [comment]
The Python test tries to load, save, and reload, each file in `tests/` and then compares them.  It looks like `intermediate.wcon` fails this test for Python 3 only because of how `0` is serialized.  I'll check it out.

**Comment by MichaelCurrie** [comment]
Error log:

```
ERROR: test_save_and_load (tests.TestWCONParser)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/openworm/tracker-commons/src/Python/tests/tests.py", line 148, in test_save_and_load
    w_loaded.save_to_file(test_path, pretty_print=pretty)
  File "../wcon/wcon_parser.py", line 416, in save_to_file
    indent=4 if pretty_print else None)
  File "/home/travis/miniconda3/lib/python3.4/json/__init__.py", line 178, in dump
    for chunk in iterable:
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 422, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 396, in _iterencode_dict
    yield from chunks
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 317, in _iterencode_list
    yield from chunks
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 396, in _iterencode_dict
    yield from chunks
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 317, in _iterencode_list
    yield from chunks
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 429, in _iterencode
    o = _default(o)
  File "/home/travis/miniconda3/lib/python3.4/json/encoder.py", line 173, in default
    raise TypeError(repr(o) + " is not JSON serializable")
TypeError: 0 is not JSON serializable
```

The issue was `intermediate.wcon` used a `0` and a `1`, integers, as time indices, which I hadn't tested for in previous examples.  My previous examples all used floats.  It's the same reason why we get this result, [as illustrated here](http://stackoverflow.com/questions/11561932/):

```
>>> json.dumps(np.arange(1,5))
*** TypeError: array([1, 2, 3, 4]) is not JSON serializable
```

Essentially, the pandas data representation returns an `int32` which JSON can't deal with since it's not a native Python data type.

The solution was in the reading step to load the timeframes as floats:

```
cur_timeframes = np.array(data_segment['t'])
```

(e.g. `[0,1]`)

becomes

```
cur_timeframes = np.array(data_segment['t']).astype(float)
```

(e.g. `[0., 1.]`)

**Comment by MichaelCurrie** [comment]
Fixed by 9b7faf064302439085980fcd982eaf64cdf8fe17
